src/lkl/setup.c

```diff
@@ -1211,45 +1252,43 @@ static void* lkl_termination_thread(void* args)
             lkl_strerror(ret));
     }
 
-    // Unmount disks
+    // Unmount mounts
     long res;
-    for (int i = num_disks - 1; i >= 0; --i)
+    for (int i = sgxlkl_enclave_state.num_disk_state - 1; i > 0; --i)
     {
-        if (!disks[i].mounted)
+        if (!sgxlkl_enclave_state.disk_state[i].mounted)
             continue;
 
-        SGXLKL_VERBOSE("calling lkl_umount_timeout(%s)\n", disks[i].mnt);
-        res = lkl_umount_timeout(disks[i].mnt, 0, UMOUNT_DISK_TIMEOUT);
+        sgxlkl_enclave_mount_config_t* disk_i = &cfg->mounts[i];
+        SGXLKL_VERBOSE("calling lkl_umount_timeout(%s)\n", disk_i->destination);
+        res = lkl_umount_timeout(disk_i->destination, 0, UMOUNT_DISK_TIMEOUT);
         if (res < 0)
         {
             sgxlkl_warn(
                 "Could not unmount disk %d, %s\n", i, lkl_strerror(res));
         }
 
-        // Root disk, no need to remove mount point ("/").
-        if (i == 0)
-            continue;
-
-        // Read-only root disk, cannot remove pre-existing mount points for
```

Comment:
What did I remove? Note that the root disk is separate now, it's unmounted after the loop.

Reply:
That:
```
        // Read-only root disk, cannot remove pre-existing mount points for
        // secondary disks.
        if (disks[0].ro)
            continue;
```

Reply:
So I shouldn't try to unmount a read-only root disk?

Reply:
Please look at the full code as it was before. If the root disk is read-only, then you cannot *remove* the mount point folders of the additional disks...because the root disk is readonly.

Reply:
Oh yeah, that makes sense, I'll change that. 

(Also, see comment below starting with "Not really necessary...")

On the side: It is at this point in time that my GitHub interface changed. I now have a "Start a review" button where the "Comment" button used to be.